include/boost/url/experimental/impl/router.hpp

```diff
@@ -0,0 +1,175 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#include <boost/url/detail/except.hpp>
+
+namespace boost {
+namespace urls {
+namespace experimental {
+
+template <class T>
+void
+router<T>::route(string_view path, T const& resource)
```

Comment:
This algorithm has to be an ordinary function (non template)

Reply:
```cpp
template <class T>
void
router<T>::route(string_view path);
```

?

Reply:
The algorithm should not depend on T